bindings/rust/src/lib.rs

```diff
@@ -79,10 +79,10 @@ pub type Value = sys::FizzyValue;
 // NOTE: the union does not have i32
 impl Value {
     pub fn as_i32(&self) -> i32 {
-        unsafe { self.i64 as i32 }
+        unsafe { self.i32 as i32 }
     }
     pub fn as_u32(&self) -> u32 {
-        unsafe { self.i64 as u32 }
+        unsafe { self.i32 }
```

Comment:
The confusing fact is that even though it is called `i32`, the type is `u32` in the bindgen translated fizzy.h.